pkg/file/joiner/joiner.go

```diff
@@ -118,20 +118,25 @@ func New(ctx context.Context, g storage.Getter, putter storage.Putter, address s
 	spanFn := func(data []byte) (redundancy.Level, int64) {
 		return 0, int64(bmt.LengthFromSpan(data[:swarm.SpanSize]))
 	}
-	var strategy getter.Strategy
-	var strict bool
-	var fetcherTimeout time.Duration
+	strategy, strict, fetcherTimeout, err := getter.GetParamsFromContext(ctx)
```

Comment:
it appears that `fetcherTimeout` 's only purpose is to allow for the testing of the feature, which imo, does not really belong in production code, so we have to test the feature in other way(s)

Reply:
well, i consider user-side testability of non-local redundancy in the production environment an advanced feature offering.